### PR TITLE
marvin: fix test_scale_vm for xenserver/Xcp-ng

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description

This PR fixes the scale VM failure noticed on XenServer/XCP-ng

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [X] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?
Ran test_scale_vm.py on a XenServer / XCP-ng env :

XCP-ng: 
```
nosetests --with-xunit --xunit-file=results.xml --with-marvin --marvin-config=/marvin/ref-trl-780-x-M7-pearl-dsilva-advanced-cfg -s -a tags=advanced --hypervisor=xenserver /marvin/tests/smoke/test_scale_vm.py
/usr/lib/python2.7/site-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.hazmat.backends import default_backend

==== Marvin Init Started ====

=== Marvin Parse Config Successful ===

=== Marvin Setting TestData Successful===

==== Log Folder Path: /marvin/MarvinLogs/Apr_15_2021_07_50_46_XWEN9N. All logs will be available here ====

=== Marvin Init Logging Successful===

==== Marvin Init Successful ====
=== TestName: test_01_scale_vm | Status : SUCCESS ===

===final results are now copied to: /marvin//MarvinLogs/test_scale_vm_LPJEUB===
[root@ref-trl-780-x-M7-pearl-dsilva-marvin ~]# 
```

Xenserver:
```
[root@ref-trl-782-x-M7-pearl-dsilva-marvin ~]# nosetests --with-xunit --xunit-file=results.xml --with-marvin --marvin-config=/marvin/ref-trl-782-x-M7-pearl-dsilva-advanced-cfg -s -a tags=advanced --hypervisor=Xenserver ./test_3.py 
/usr/lib/python2.7/site-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.hazmat.backends import default_backend

==== Marvin Init Started ====

=== Marvin Parse Config Successful ===

=== Marvin Setting TestData Successful===

==== Log Folder Path: /marvin/MarvinLogs/Apr_21_2021_07_19_49_C4IRNH. All logs will be available here ====

=== Marvin Init Logging Successful===

==== Marvin Init Successful ====
====Trying SSH Connection: Host:10.0.33.228 User:root                                   Port:22 RetryCnt:10===
===SSH to Host 10.0.33.228 port : 22 SUCCESSFUL===
===final results are now copied to: /marvin//MarvinLogs//Y51I1H===
```

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 xenserver-71

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 xcpng76

File: plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java [hunk 1]
```diff
@@ -3110,7 +3110,7 @@ private long getStaticMax(final String os, final boolean b, final long dynamicMi
             s_logger.warn("No recommended value found for dynamic max, setting static max and dynamic max equal");
             return dynamicMaxRam;
         }
-        final long staticMax = Math.min(recommendedValue, 4l * dynamicMinRam); // XS
+        final long staticMax = Math.max(recommendedValue, 4L * dynamicMinRam); // XS
```
Comment (harikrishna-patnala) [inline plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java:3113]: 4 times dynamicMinRam can be higher than the recommended value, so it needs be min() only.

File: test/integration/smoke/test_scale_vm.py [hunk 2]
```diff
@@ -91,6 +99,7 @@ def setUpClass(cls):
             mode=cls.services["mode"]
         )
         cls._cleanup = [
```
Comment (DaanHoogland) [inline test/integration/smoke/test_scale_vm.py:101]: can we do a "proper" cleanup construct here and not expand on this one, now that we are touching this file? see `smoke.test_accounts.TestRemoveUserFromAccount` as example.

**Review by DaanHoogland — COMMENTED** [review]
great if we finally tackle this. @harikrishna-patnala 's comment worries me a bit.

**Comment by Pearl1594** [comment]
@DaanHoogland @harikrishna-patnala choosing the max value may actually not be ideal for the reason mentioned by Hari. Should we maybe skip this test for xenserver?

**Comment by DaanHoogland** [comment]
@Pearl1594 do you mean for xenserver only or for xcpng as well?
I always thought that the failure was due to licensing, in which case it makes sense to skip for xenserver only.

**Comment by Pearl1594** [comment]
yes @DaanHoogland - skip it only for XenServer. It works fine for XCP-ng

**Comment by harikrishna-patnala** [comment]
Yes @Pearl1594, if we cannot do it without licensed XenServers then I vote for skipping it for XenServer and keeping it for XCP

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 xenserver-71

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 xcpng76

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 xenserver-71

**Comment by Pearl1594** [comment]
@blueorangutan test centos7 xcpng76

File: server/src/main/java/com/cloud/vm/UserVmManagerImpl.java [hunk 3]
```diff
@@ -1946,12 +1946,6 @@ private boolean upgradeRunningVirtualMachine(Long vmId, Long newServiceOfferingI
...
                     success = true;
                     return success;
-                } catch (InsufficientCapacityException e) {
-                    s_logger.warn("Received exception while scaling ", e);
-                } catch (ResourceUnavailableException e) {
-                    s_logger.warn("Received exception while scaling ", e);
-                } catch (ConcurrentOperationException e) {
-                    s_logger.warn("Received exception while scaling ", e);
                 } catch (Exception e) {
                     s_logger.warn("Received exception while scaling ", e);
```
Comment (shwstppr) [inline server/src/main/java/com/cloud/vm/UserVmManagerImpl.java:1950]: @Pearl1594 do we need to catch all exceptions here?

Comment (shwstppr) [reply server/src/main/java/com/cloud/vm/UserVmManagerImpl.java]: @Pearl1594 sorry for the confusion but I was referring `catch (Exception e)`. Do we really need that?
Anyway, not your change so I'm LGTM for this PR

File: test/integration/smoke/test_scale_vm.py [hunk 4]
```diff
@@ -90,17 +116,19 @@ def setUpClass(cls):
...
             mode=cls.services["mode"]
         )
-        cls._cleanup = [
-            cls.small_offering,
-            cls.account
-        ]
 
     @classmethod
     def tearDownClass(cls):
         cls.apiclient = super(
```
Comment (DaanHoogland) [inline test/integration/smoke/test_scale_vm.py:122]: cls.apiclient is already set in setup. we can remove this.

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Review by shwstppr — APPROVED** [review]
LGTM. Test passing on XCP-ng and the latest travis test for test_scale_vm also passing, https://travis-ci.com/github/apache/cloudstack/jobs/500760997

**Comment by rhtyd** [comment]
@blueorangutan test centos7 xcpng76

**Comment by rhtyd** [comment]
lgtm

**Pull request merged**